input/Shared/switches.txt

```diff
@@ -32,3 +32,11 @@
 
 All switches not recognized by Cake will be added to an argument list that is passed to the build script.
 See [Arguments And Environment Variables](/docs/writing-builds/args-and-environment-vars#arguments) how to read arguments in your script.
+
+:::{.alert .alert-info}
+Use a space instead of `=` if the value of a switch can be empty or null:
```

Comment:
Is "empty or null" precice enough? My tests always result in `null`, regardless whether I set `--foo` or `--foo ""`.

Reply:
I think it's clear, and the code example after that helps. How would phrase it differently?

Also, what do you mean by "_my tests always result in `null`_"

In my tests, both `--foo` and `--foo ""` results in the argument not being present in the collection of arguments, which means that `Argument<string>("foo")` throws an exception on both cases with `Argument 'foo' was not set`.

So the only way to get `null` is to specify `null` as the default value i.e. `Argument<string>("foo", null)` but in that case you're only getting `null` because you said that was the default... Not because Cake decided for you. The same way that `Argument<string>("foo", "abc")` will give you `abc` for both `--foo` and `--foo ""`